bin/generate

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env ruby
+
+# This script fetches the canonical test data for an exercise
+# from the web and converts it to a Vader file.
+#
+#   $ bin/generate triangle > triangle.vader
+
+require 'open-uri'
+require 'json'
+
+if RUBY_VERSION !~ /^2\.[3-9]/
+  puts "Sorry, but you need Ruby 2.3+ because squiggly heredocs are awesome."
+  exit 1
+end
+
+if ARGV[0].nil?
+  puts "usage: #{$0} <exercise slug>"
+  exit 1
+end
+
+dataurl="https://raw.githubusercontent.com/exercism/x-common/master/exercises/#{ARGV[0]}/canonical-data.json"
+
+begin
+  data = open(dataurl).read
+rescue OpenURI::HTTPError => err
+  puts "Fetching canonical data failed: #{err}"
+  puts "-> #{dataurl}"
+  exit 1
+end
+
+begin
+  data = JSON.parse(data)
+rescue JSON::ParserError => err
+  puts err
+  puts "Parsing JSON data failed."
+  exit 1
+end
+
+def header(version)
+  <<~HEREDOC
+    "
+    " Version: #{version}
+    "
+
+    Before:
+      unlet! input expected
+
+  HEREDOC
+end
+
+def ruby_hash_to_vim(hash)
+  '{' + hash.map { |k,v| "\"#{k}\": #{v.inspect}" }.join(', ') + '}'
+end
+
+def setvar(var, value)
+  s = "  let #{var} = "
+  case value
+  when NilClass
+    s << "''"
+  when TrueClass
+    s << '1'
+  when FalseClass
+    s << '0'
+  when Hash
+    s << ruby_hash_to_vim(value)
+  else
+    s << value.inspect
+  end
+end
+
+def funcname(name)
+  name[0].capitalize + name[1..-1]
+end
+
+# Unfortunately the variable name containing the test input differs
+# from exercise to exercise. Remove known keys and hope for the best
+# and hope there's only one key left...
```

Comment:
This is the only part I'm upset about, because I'm essentially guessing.

IMHO, it would be better when all canonical tests would use the same keys, `input` and `expected`. The former isn't always available. :<